overlord/snapstate/handlers_components.go

```diff
@@ -291,6 +292,15 @@ func (m *SnapManager) doMountComponent(t *state.Task, _ *tomb.Tomb) (err error)
 	perfTimings.Save(st)
 	st.Unlock()
 
+	// if we're removing the snap file and we are mounting a component for the
+	// firs time, then we know that the component also must be coming from an
+	// emphemeral file. in that case, remove it.
+	if snapsup.RemoveSnapPath {
```

Comment:
Maybe we should add the flags from `componentInstallFlags` to `ComponentSetup` and check that instead?

Reply:
Done in fef69533116301a41398f18f824541c18173eddd.

Reply:
but this is code still the same? what am I missing

Reply:
Sorry, missed that in the commit.